react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { currentIndex: 0 };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(event) {
+    if (event.target.closest('div').classList.contains('arrow-left')) {
+      this.setState({ currentIndex: ((this.state.currentIndex + this.props.items.length) - 1) % this.props.items.length });
+      this.componentWillUnmount();
```

Comment:
While this technically works we don't want to directly call the lifecycle methods ourselves. It is better to make your own methods and use those instead.